### PR TITLE
Add basic mgo interop and tests

### PR DESCRIPTION
**Pull request by kofalt**
This adds the basic capability to interact with the database using the SDK structs, which is convenient for some backend scripts. Along the way it looks like there is a newer fork of the mgo library, which I switched to.

The tests are not exhaustive, but all the structs are updated and the main hierarchy + jobs are covered.

### CODE REVIEW

File: tests/0-init_test.go [hunk 1]
```diff
@@ -82,77 +82,67 @@ type F struct {
...
+	}
 
-		client = api.NewApiKeyClient(key, opts...)
-	} else {
-		panic("Unsupported test mode " + mode)
+	if root {
+		opts = append(opts, api.EnableRoot)
 	}
 
+	client = api.NewApiKeyClient(key, opts...)
```
Comment (ehlertjd) [inline tests/0-init_test.go]: Could probably just return `api.NewApiKeyClient(...)` - I see no need for the intermediate instance anymore.

File: tests/project_test.go [hunk 2]
```diff
@@ -278,8 +278,12 @@ func (t *F) createTestProject() (string, string) {
 	groupId := t.createTestGroup()
 
 	project := &api.Project{
-		Name:    RandString(),
-		GroupId: groupId,
+		Name:        RandString(),
+		GroupId:     groupId,
+		Description: "This is a description",
```
Comment (ehlertjd) [inline tests/project_test.go]: Does this change serve a purpose?

File: tests/session_test.go [hunk 3]
```diff
@@ -230,6 +230,19 @@ func (t *F) createTestSession() (string, string, string) {
 	session := &api.Session{
 		Name:      sessionName,
 		ProjectId: projectId,
+		Info: map[string]interface{}{
```
Comment (ehlertjd) [inline tests/session_test.go]: Does this change serve a purpose?

**Comment by ryansanford** [comment]
What are the use cases, beyond injecting user API keys?

**Comment by kofalt** [comment]
It's not very interesting, unless you're using mgo like the `.circleci/*-environment.go` scripts.
Otherwise, you get stuck with type annotations:

```go
elapsed1 := item["profile"].(map[string]interface{})["elapsed"].(int)
```

**Pull request merged**